### PR TITLE
Wrap final heartbeat message in a try except block

### PR DESCRIPTION
**Pull request by savingoyal**

### CODE REVIEW

File: metaflow/metadata/heartbeat.py [hunk 1]
```diff
@@ -53,5 +52,9 @@ def heartbeat(self):
             return response.json().get('wait_time_in_seconds')
 
     def shutdown(self):
-        # attempts sending on last heartbeat
-        self.heartbeat()
+        # attempts sending one last heartbeat
+        try:
+            self.heartbeat()
+        except:
```
[the remark this answers is not in the document]
Comment (savingoyal) [reply metaflow/metadata/heartbeat.py:58]: for sure!

[the remark this answers is not in the document]
Comment (savingoyal) [reply metaflow/metadata/heartbeat.py:58]: @ferras I don't think the [stacktrace print](https://github.com/Netflix/metaflow/blob/master/metaflow/sidecar_worker.py#L38) is bubbled up to the end user

[the remark this answers is not in the document]
Comment (savingoyal) [reply metaflow/metadata/heartbeat.py:58]: #518 superceded this PR. I will copy this comment there.

[the remark this answers is not in the document]
Comment (savingoyal) [reply metaflow/metadata/heartbeat.py:58]: The `ping` method swallows the exception so there is nothing for the heartbeat process to report.

[the remark this answers is not in the document]
Comment (ferras) [reply metaflow/metadata/heartbeat.py:58]: Note in case someone checks this thread later: we only redirect stdout to devnull, not stderr

**Pull request closed without merging**